### DISCUSSION TITLE
How to set focus on page load?

### DISCUSSION THREAD

**Discussion by tsongas** [react-hook-form/react-hook-form#5130] [Q&A]
I'm starting to learn react-hook-form and am wondering how to set focus on the username field of a login page when the page loads. I didn't see an example here https://github.com/react-hook-form/react-hook-form/tree/master/examples. I did find this generic example on StackOverflow:

```
const { useRef, useEffect } = React;

function Foo() {
  const txt1 = useRef(null);

  useEffect(() => {
    txt1.current.focus();
  }, []);

  return <input type="text" ref={txt1} />;
}

ReactDOM.render(<Foo />, document.getElementById("root"));
```

however I'm just getting familiar with refs and am not sure how to use them with react-hook-form to accomplish this with `useEffect()`. I'm also using Chakra-UI, here's a CodeSandbox example I set up https://codesandbox.io/s/reverent-lalande-0elrt?file=/src/App.tsx.

**Comment by tsongas** [comment]
I just added `autoFocus={true}` to the Input and that is working in the CodeSandbox, but it doesn't work in my actual project. I'll have to investigate further and post a follow-up.

**Reply by bluebill1049** [reply]
Look up for setFocus method.
